src/native/s3_client.c

```diff
@@ -523,10 +525,28 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientMake
         resume_token = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_resume_token);
     }
 
+    struct aws_s3_checksum_config checksum_config = {
+        .location = checksum_location,
+        .checksum_algorithm = checksum_algorithm,
+        .validate_response_checksum = validate_response,
+    };
+
+    struct aws_array_list response_checksum_list;
```

Comment:
you always clean it up in done, but there are already jumps into done before this declaration. cleaning up undeclared variable will be a segfault